src/scripts/shards/Shards.ts

```diff
@@ -0,0 +1,98 @@
+class Shards implements Feature {
+    name: string = "Shards";
+    saveKey: string = "shards";
+
+    defaults = {
+        'shardWallet': Array.apply(null, Array<number>(18)).map(() => 0),
+        'shardUpgrades': Array.apply(null, Array<number>(18 * 4)).map(() => 0),
```

Comment:
Why not an 18 * 4 array?

Reply:
Our `ArrayOfObservables` is designed to work on 1d arrays. I didn't want to spend the time to make a 2D array observe correctly.